### PR TITLE
increasing array queries

### LINKED ISSUE #4091: Contact Form Submission - Unclear Explanation (CSES - Increasing Array Queries)
Someone submitted the contact form!

**URL**: https://usaco.guide/problems/cses-2416-increasing-array-queries/solution?lang=cpp
**Module**: CSES - Increasing Array Queries
**Topic**: Unclear Explanation
**Message**: 
Unclear explanation. For example- what is mx0 or mxj. Similarly many other things are mentioned without their definition.

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

closes #4091

### CODE REVIEW

File: solutions/gold/cses-2416.mdx [hunk 1]
```diff
@@ -2,106 +2,185 @@
...
 
 $$
-(\texttt{idx}_{j + 1} - 1 - \texttt{idx}_j) \cdot \texttt{mx}_j - (\texttt{pref}_{\texttt{idx}_{j + 1} - 1} - \texttt{pref}_{\texttt{idx}_j})
+x \cdot (j-i) - \sum_{k=i}^{j-1} \texttt{arr}[k]
 $$
 
-This means that for a fixed $l$, we can compute each $\texttt{mx}_j$ and their
-contributions, and then use `std::upper_bound` and a BIT to answer queries with
-variable $r$!
+The second term is the sum of all elements in the interval $[i,j-1)$.
```
Comment (devo1ution) [inline solutions/gold/cses-2416.mdx:49]: Should this be [i, j)? (the summation goes from i to j - 1 right)

File: solutions/gold/cses-2416.mdx [hunk 2]
```diff
@@ -2,106 +2,185 @@
...
+Can you first come up with an algorithm for a single query?
+It doesn't have to be crazy efficient; $\mathcal{O}(N)$ time is fine.
+
+</Spoiler>
+
+<Spoiler title="Answer to Hint 1">
+
+Let's iterate through the array in order and keep track of the largest element $m$ we've seen so far.
+For each element $x$, we first update $m$.
+Then, if the current one is smaller than $m$, we add $m-x$ to the total
```
Comment (devo1ution) [inline solutions/gold/cses-2416.mdx:19]:
```suggestion
Then, if the current one is smaller than $m$, we add $m-x$ to the total.
```

File: solutions/gold/cses-2416.mdx [hunk 3]
```diff
@@ -2,106 +2,185 @@
...
-variable $r$!
+The second term is the sum of all elements in the interval $[i,j-1)$.
 
-If we want to change $l$, notice how we can update $\texttt{mx}$ efficiently by
-using a monotone stack. Since each $\texttt{mx}_j$'s contribution depends on
-$x_i$ **after** it, the contributions of the old $\texttt{mx}_j$ don't change.
+For example, in that array in Hint 2, $10$ contributes $10 \cdot (2-0)-(10+4)=6$, while $11$ contributes $11 \cdot (4-2)-(11+3)=8$ to the sum.
 
-Again, we use `std::upper_bound` and a BIT to answer queries.
+Notice that this doesn't apply when an element's $j$ goes past the end of the query.
```
Comment (devo1ution) [inline solutions/gold/cses-2416.mdx:53]:
```suggestion
Notice that this doesn't apply when an element's $j$ goes past the end of the query;
```

File: solutions/gold/cses-2416.mdx [hunk 4]
```diff
@@ -2,106 +2,185 @@
...
+How do we utilize this?
 
-```cpp
-#include <bits/stdc++.h>
-typedef long long ll;
-using namespace std;
+First, we have to iterate through the array in reverse.
 
-const ll INF = 1e18;
+Let's also keep track of a stack of the values and indicies of all the maximums one would get when iterating from an index $i$ to the end of the array.
```
Comment (devo1ution) [inline solutions/gold/cses-2416.mdx:62]:
```suggestion
Let's also keep track of a stack of the values and indices of all the maximums one would get when iterating from an index $i$ to the end of the array.
```

File: solutions/gold/cses-2416.mdx [hunk 5]
```diff
@@ -2,106 +2,185 @@
...
 
-int n, q;
-ll x[200002], pref[200002], ans[200002];
-ll contrib[200002], bit[200002];
-vector<pair<int, int>> bckt[200001];
+Taking the array from Hint 2 again. If we were at index $1$, our stack would be $[(11, 2), (4, 1)]$, with the end of the array being the top of the stack.
```
Comment (devo1ution) [inline solutions/gold/cses-2416.mdx:64]:
```suggestion
Taking the array from Hint 2 again, If we were at index $1$, our stack would be $[(11, 2), (4, 1)]$, with the end of the array being the top of the stack.
```

**Review by devo1ution — CHANGES_REQUESTED** [review]

**Comment by devo1ution** [comment]
w editorial 👍🏻

File: solutions/gold/cses-2416.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/gold/cses-2416.mdx:49]: oh yeah mbmb

File: solutions/gold/cses-2416.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/gold/cses-2416.mdx:53]: does this matter that much?
i can change it if you insist

Comment (devo1ution) [reply solutions/gold/cses-2416.mdx:53]: > does this matter that much? i can change it if you insist
nah not really lol it js reads better for me imo but its really minor

**Review by devo1ution — APPROVED** [review]

File: solutions/gold/cses-2416.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/gold/cses-2416.mdx:53]: alright
if anyone reads this and agrees with daniel, just tell me and i'll change

**Review by ryanchou-dev — APPROVED** [review]
amazing

**Pull request merged**